native/core/src/execution/shuffle/shuffle_writer.rs

```diff
@@ -1650,18 +1719,44 @@ mod test {
 
     #[test]
     #[cfg_attr(miri, ignore)] // miri can't call foreign function `ZSTD_createCCtx`
-    fn write_ipc_zstd() {
+    fn roundtrip_ipc_zstd() {
         let batch = create_batch(8192);
         let mut output = vec![];
         let mut cursor = Cursor::new(&mut output);
-        write_ipc_compressed(
+        let length = write_ipc_compressed(
             &batch,
             &mut cursor,
             &CompressionCodec::Zstd(1),
```

Comment:
Could we modify this test so instead of being named `round_trip_zstd` it iterates through all the compression schemes?

Reply:
Good idea. I have implemented this in https://github.com/apache/datafusion-comet/pull/1192/commits/1fc3d4998a355f84c31953ae197a707933539d25 and also fixed the bug that it exposed :slightly_smiling_face: